### ISSUE TITLE
Modify Environment Variables http_proxy and https_proxy have Dart specific names

### ISSUE THREAD

**Issue by The-Funk** [dart-lang/pub#3492]
I'd like the pub http_proxy and https_proxy environment variable names found [here](https://dart.dev/tools/pub/troubleshoot) to become something more Dart specific, like dart_http_proxy and dart_https_proxy. As the environment variable at the moment is very open ended in name, other services have tried to use it and failed because of unexpected formatting. For instance, the azure cli. Azure command line tools will attempt to use the https_proxy environment variable if found. This causes the az login command to fail and breaks some of my pipelines.

The only workaround is to constantly create and delete the environment variable depending on if I'm getting dependencies for a Flutter project or pushing a container to Azure, respectively.

**Comment by sigurdm** [comment]
Is the problem that we expect these variables in a different format from other software?

cc @jonasfj

**Comment by The-Funk** [comment]
Yes exactly. The http_proxy/https_proxy variables for azure cli are expected to be provided in a different format and as such if I set the variable for azure cli then I can't download packages with pub get and if I change the variables to provide data in the Dart format then I can't access azure cli. This is just one example, I'm sure there are other applications in the wild that use these same environment variables. Maybe, to prevent a breaking change, you could support both http_proxy/https_proxy and a new set of variables dart_http_proxy/dart_https_proxy and put priority on the dart specific variable names so that if both are provided, the dart names are consulted first. Then just update docs to reflect the new variable names. If you'd like, I could probably write that code and do the PR.

**Comment by sigurdm** [comment]
The support for these environment variables are implemented in the sdk:(https://dart.googlesource.com/sdk/+/e5af733398d3d915dd794f765268c33428dd5729/sdk/lib/_http/http.dart#1501), not in pub.

Do you have documentation on the format of http_proxy for the azure cli? Not sure I understand the difference.

**Comment by The-Funk** [comment]
> The support for these environment variables are implemented in the sdk:(https://dart.googlesource.com/sdk/+/e5af733398d3d915dd794f765268c33428dd5729/sdk/lib/_http/http.dart#1501), not in pub.
> 
> Do you have documentation on the format of http_proxy for the azure cli? Not sure I understand the difference.

I actually just checked and they use the exact same format. I thought the issue was that the format was different, the actual issue is that only Dart needs the HTTP_PROXY variable set. When I was changing to another format, the value was being ignored by az cli, and thus az cli functioned normally. I think regardless of format this is an important change to consider for those of us working in corporate environments but I suppose it's a change for the SDK.

**Comment by sigurdm** [comment]
Closing as obsolete.
